pkg/configuration/publicviewer_config.go

```diff
@@ -0,0 +1,25 @@
+package configuration
+
+import (
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+)
+
+type PublicViewerConfig struct {
+	Config toolchainv1alpha1.PublicViewerConfig
+}
+
+func (c PublicViewerConfig) Enabled() bool {
+	return c.Config.Enabled
+}
+
+func (c PublicViewerConfig) Username() string {
+	return c.Config.Username
+}
+
+func (c PublicViewerConfig) IsPublicViewer(username string) bool {
+	if !c.Enabled() {
+		return false
+	}
+
+	return c.Config.Username == username
+}
```

Comment:
should we check that c.Config.Username is not empty ? I mean could it happen that we enable the config without setting a username? WDYT?

Reply:
the RegEx validation should prevent this from happening

```
// +kubebuilder:validation:Pattern=`^[a-z0-9]([-a-z0-9]*[a-z0-9])?$`
```

Reply:
no, you're right, this can happen. Let me look into it better, great catch!

Reply:
Do you think [this approach](https://github.com/codeready-toolchain/host-operator/pull/998#discussion_r1537753050) could help also here?

Reply:
As pointed out in the comment https://github.com/codeready-toolchain/host-operator/pull/998#discussion_r1538759675 , I would remove the field.